File: app/src/main/java/ru/rinet/questik/data/room/dao/CorpDao.java
package ru.rinet.questik.data.room.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import ru.rinet.questik.data.room.models.Corp;

import static androidx.room.OnConflictStrategy.REPLACE;

@Dao
public interface CorpDao {
    //запрос всей таблицы
    @Query("SELECT * FROM Corp")
    List<Corp> getAll();

    //запрос на каегорию по его id
    @Query("SELECT * FROM Corp WHERE corp_id = :id")
    long getById(Long id);

    @Query("SELECT * FROM Corp")
    Corp getCorp();

    @Query("SELECT * FROM Corp")
    List<Corp> getCorpList();

    //Wipe
    @Query("DELETE FROM Corp")
    void wipeCorpTable();

    @Insert(onConflict = REPLACE)
    Long insert(Corp corp);

    @Insert
    List<Long> insertCarpList(List<Corp> corpList);

    @Update
    int update(Corp corp);

    @Delete
    int delete(Corp corp);


}

File: QuestikCompose/app/src/main/java/xyz/gmfatoom/questik/repo/local/room/dao/CorpDao.kt
package xyz.gmfatoom.questik.repo.local.room.dao

import androidx.room.*
import kotlinx.coroutines.flow.Flow
import xyz.gmfatoom.questik.repo.local.room.entity.CorpEntity


@Dao
interface CorpDao {
    @Query("SELECT COUNT(corp_id) FROM corp")
    fun getCorpCount(): Int
    @Query("SELECT COUNT(corp_id) FROM corp")
    fun getCorpCountFlow(): Flow<Int>
/*    @Query("SELECT * FROM contacts")
    fun getAllCorp(): PagingSource<Int, CorpEntity>*/

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAllCorp(items:List<CorpEntity>)

    @Query("DELETE FROM corp")
    fun clearCorp()

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertCorp(corp: CorpEntity):Long

    @Update
    fun updateCorp(corp: CorpEntity)

    @Delete
    fun deleteCorp(corp: CorpEntity)

    @Query("SELECT * FROM corp WHERE corp_id == :id")
    fun getCorpById(id: Int): CorpEntity

    @Query("SELECT * FROM corp")
    fun getCorp(): List<CorpEntity>
}